ui.go

```diff
@@ -468,8 +468,11 @@ func (win *win) printDir(screen tcell.Screen, dir *dir, context *dirContext, dir
 
 		filename := []rune(f.Name())
 		if runeSliceWidth(filename) > maxFilenameWidth {
-			filename = runeSliceWidthRange(filename, 0, maxFilenameWidth-1)
+			truncatePos := (maxFilenameWidth - 1) * gOpts.truncatepct / 100
+			var lastPart []rune = runeSliceWidthLastRange(filename, maxFilenameWidth-truncatePos-1)
```

Comment:
`runeSliceWidthLastRange` already returns `[]rune` so you should just be able to write this:

```suggestion
			lastPart := runeSliceWidthLastRange(filename, maxFilenameWidth-truncatePos-1)
```